app/models/loan.rb

```diff
@@ -1,20 +1,18 @@
 class Loan < ActiveRecord::Base
   include TranslationModule, MediaModule
 
-  belongs_to :cooperative, :foreign_key => 'CooperativeID'
-  belongs_to :division, :foreign_key => 'SourceDivision'
+  belongs_to :division
+  belongs_to :primary_agent, class_name: 'Person'
+  belongs_to :secondary_agent, class_name: 'Person'
+  belongs_to :currency
+  belongs_to :representative, class_name: 'Person'
```

Comment:
It has `organization_id` as one of the fields. I think there is always one organization that receives a loan. I don't see the business logic for a many-to-many.